### ISSUE TITLE
New principle: Web IDL is for APIs exposed to the web by browser engines

### ISSUE THREAD

**Issue by hober** [w3ctag/design-principles#216]
(Splitting this off from #128)

The design principles document currently says:

> The language that web APIs are meant to be used in, and specified for, is JavaScript (also known as ECMAScript). They are not language-agnostic, and are not meant to be.

I think there's a corresponding principle, which is that the platform that Web IDL is meant to be used for is the web as implemented by browser engines. We should discourage spec authors from using Web IDL to define interfaces not intended for wide deployment in browser engines.

**Comment by swickr** [comment]
> We should discourage spec authors from using Web IDL to define interfaces not intended for wide deployment in browser engines.

One way to discourage this usage is to recommend what IDL(s) _are_ preferred for specifications of interfaces intended to be used primarily in other than browser engines.

**Comment by annevk** [comment]
This might dissuade the JavaScript and WebAssembly communities from adopting IDL as a way of defining their APIs. I don't think that's desirable as we could use the help in maintaining it.

**Comment by gkellogg** [comment]
I hope that this doesn't come down to recommending multiple different IDL formats to do essentially the same thing. JSON-LD used WebIDL, partly due to no other alternative that met the requirements, but also because it adequately and unambiguously allows us to describe the interfaces to describe implementation requirements, even if non-JavaScript implementations might interpret something like Promises differently. The evidence is that it leads to good interoperability between implementations, which, after all, is the whole point of specifying an interface.

**Comment by annevk** [comment]
Yeah, I also tend to think it's fine to use IDL for other things, as long as `[Exposed]` is used correctly it will be pretty clear who is expected to implement it.
